### ISSUE TITLE
Support `await`ing messages

### ISSUE THREAD

**Issue by philpax** [AmbientRun/Ambient#710]
At present, `Message`s require you to set up a subscription that lasts until you cancel it, which is not the most ergonomic for one-shot messages or for a back-and-forth protocol.

Instead, we should
- offer the ability to one-shot subscribe to a message (this existed previously, but was removed for Reasons:tm:)
- use the former to provide a nice async wrapper for awaiting messages:

```rust
let (source, value) = MyCoolMessage::receive().await;
```

(it feels to me that `(value, source)` is a better order, but `(source, value)` is consistent with `subscribe`)

Also relevant: 
- #673 
- #253

**Comment by chaosprint** [comment]
Is this related to https://github.com/AmbientRun/Ambient/issues/665

**Comment by philpax** [comment]
Nah, this is more of an API detail about how you use messages than how the messages are sent.
